samples/boards/olimex_stm32_e407/ccm/README.rst

```diff
@@ -0,0 +1,188 @@
+.. _olimex_stm32_e407_ccm:
+
+Olimex STM32-E407 CCM example
+#############################
+
+Overview
+********
+
+Show usage of the Core Coupled Memory (CCM) that is available
+on several STM32 devices. The very important difference with
+normal RAM is that CCM can not be used for DMA.
+
+By prefixing a variable with __ccm_data_section, __ccm_bss_section,
+or __ccm_noinit_section those variables are placed in the CCM.
+
+The __ccm_data_section prefix should be used for variables that
+are initialized. Like the normal data section the initial
+values take up space in the FLASH image.
+
+The __ccm_bss_section prefix should be used for variables that
+should be initialized to 0. Like the normal bss section they
+do not take up FLASH space.
+
+The __ccm_noinit_section prefix should be used for variables
+that don't need to have a defined initial value (for example
+buffers that will receive data). Compared to bss or data the
+kernel does not need to initialize the noinit section making
+the startup slightly faster.
+
+To add CCM support to a board, add the following line to the
+board's DTS file ``chosen`` section:
+
+.. code-block:: console
+
+   zephyr,ccm = &ccm0;
+
+For example the olimex STM32 E407 DTS file looks like this:
+
+.. code-block:: console
```

Comment:
To avoid having the doc go out of sync with the dts file, you can just include it here with (leave off the linenos parameter if you don't want line numbers displayed):

     .. literalinclude:: ../../olimex_stm32_e407.dts
        :linenos: